mlflow/store/dbfs_artifact_repo.py

```diff
@@ -28,12 +28,14 @@ def __init__(self, artifact_uri, get_host_creds):
         if not cleaned_artifact_uri.startswith('dbfs:/'):
             raise MlflowException('DbfsArtifactRepository URI must start with dbfs:/')
 
-    def _databricks_api_request(self, **kwargs):
+    def _databricks_api_request(self, endpoint, **kwargs):
         host_creds = self.get_host_creds()
-        return http_request(host_creds, **kwargs)
+        return http_request_safe(host_creds=host_creds, endpoint=endpoint, **kwargs)
 
     def _dbfs_list_api(self, json):
-        return self._databricks_api_request(endpoint=LIST_API_ENDPOINT, method='GET', json=json)
+        host_creds = self.get_host_creds()
+        return http_request(
```

Comment:
As in `databricks.py`, here we still use the old `http_request` API since non-200 response statuses can indicate that no files exist under the specified DBFS path